smarts/core/mission_planner.py

```diff
@@ -299,12 +300,28 @@ def cut_in_waypoints(self, sim, pose: Pose, vehicle, base_waypoint_generator):
     def uturn_waypoints(self, sim, pose: Pose, vehicle):
         # TODO: 1. Need to revisit the approach to calculate the U-Turn trajectory.
         #       2. Wrap this method in a helper.
+
+        ego_position = pose.position[:2]
+        ego_lane = self._road_network.nearest_lane(ego_position)
+        ego_wps = self._waypoints.waypoint_paths_on_lane_at(
+            ego_position, ego_lane.getID(), 60
+        )
+        if self._mission.task.initial_speed is None:
+            default_speed = ego_wps[0][0].speed_limit
+        else:
+            default_speed = self._mission.task.initial_speed
+        ego_wps_des_speed = []
+        for px in range(len(ego_wps[0])):
+            new_wp = replace(ego_wps[0][px], speed_limit=default_speed)
+            ego_wps_des_speed.append(new_wp)
+
+        ego_wps_des_speed = [ego_wps_des_speed]
```

Comment:
Is this `ego_wps_des_speed` variable meant to have two nested lists `[[ ... ]]` ? Is the outer list necessary?

Reply:
Thanks, Yes, exactly, it should be a list of lists, otherwise it throws an error.